File: 250129/최고의 33위치/best-place-of-33.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
        // Please write your code here.
        Scanner scanner = new Scanner(System.in);
        int n = scanner.nextInt();
        int[][] grid = new int[n][n];
        for(int i = 0; i < n; i++){
            for(int j = 0; j < n; j++){
                grid[i][j] = scanner.nextInt();
            }
        }
        int[] dx = {-1, 0, 1, -1, 0, 1, -1, 0, 1};
        int[] dy = {1, 1, 1, 0, 0, 0, -1, -1, -1};
        int max = 0;
        for(int i = 1; i < n-1; i++){
            for(int j = 1; j < n-1; j++){
                int count = 0;
                for(int k = 0; k < 9; k++){
                    if(grid[i + dy[k]][j + dx[k]] == 1){
                        ++count;
                    }
                }
                if(count > max) max = count;
            }
        }
        System.out.print(max);
    }
}

File: 251118/최고의 33위치/best-place-of-33.py
n = int(input())
grid = [list(map(int, input().split())) for _ in range(n)]

# Please write your code here.
dx = [-1, 0, 1]
dy = [-1, 0, 1]

maxNum = 0
for r in range(n):
    for c in range(n):
        curr = 0;
        for xx in dx:
            for yy in dy:
                x = r + xx
                y = c + yy
                if x >= 0 and x < n and y >= 0 and y < n:
                    curr += grid[x][y]
        maxNum = max(maxNum, curr)
    
print(maxNum)